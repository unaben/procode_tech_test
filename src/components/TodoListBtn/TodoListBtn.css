.btn-area {
  margin-bottom: 1rem;
}

.secondary-btn {
  background: rgb(71, 71, 71);
  color: white;
  border: none;
  margin-top: 1.6rem;
  padding: 0.7em;
  width: fit-content;
  cursor: pointer;
}

.active {
  background-color: var(--clr-primary-bg);
}

.todo-list {
  display: flex;
  flex-direction: column;
}
