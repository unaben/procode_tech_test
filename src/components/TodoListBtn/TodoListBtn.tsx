import React, { FC } from "react";
import { message } from "../../translation/ENT";
import "./TodoListBtn.css";

type ITodoListBtnProps = {
  toggleTodoBtn: boolean;
  setToggleTodoBtn: React.Dispatch<React.SetStateAction<boolean>>;
};

const TodoListBtn: FC<ITodoListBtnProps> = ({
  setToggleTodoBtn,
  toggleTodoBtn,
}) => {
  return (
    <div className="btn-area">
      <button
        onClick={() => setToggleTodoBtn(false)}
        className={`secondary-btn ${toggleTodoBtn === false && "active"}`}
      >
        {message.pending}
      </button>
      <button
        onClick={() => setToggleTodoBtn(true)}
        className={`secondary-btn ${toggleTodoBtn === true && "active"}`}
      >
        {message.completed}
      </button>
    </div>
  );
};

export default TodoListBtn;
