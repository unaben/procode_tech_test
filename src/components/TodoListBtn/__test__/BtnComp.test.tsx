import { render, screen } from "@testing-library/react";
import TodoListBtn from "../TodoListBtn";
import { Provider } from "react-redux";
import { store } from "../../../app/store";
import userEvent from "@testing-library/user-event";

describe("Btn component", () => {
  it("should renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <TodoListBtn toggleTodoBtn={false} setToggleTodoBtn={jest.fn()} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should renders button correctly", async () => {
    render(
      <Provider store={store}>
        <TodoListBtn toggleTodoBtn={false} setToggleTodoBtn={jest.fn()} />
      </Provider>
    );
    const btnEl = await screen.findAllByRole("button");
    expect(btnEl).toHaveLength(2);
  });

  it("should be called when click", async () => {
    const mockFunction = jest.fn(() => console.info("button clicked"));
    render(
      <Provider store={store}>
        <TodoListBtn toggleTodoBtn={false} setToggleTodoBtn={mockFunction} />
      </Provider>
    );
    const btnEl = screen.getByRole("button", {
      name: /Todo/i,
    });
    await userEvent.click(btnEl);
    expect(mockFunction).toHaveBeenCalledTimes(1);
  });
});
