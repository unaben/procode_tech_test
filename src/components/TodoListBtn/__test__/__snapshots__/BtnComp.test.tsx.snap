// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn component should renders correctly 1`] = `
<div>
  <div
    class="btn-area"
  >
    <button
      class="secondary-btn active"
    >
      Pending
    </button>
    <button
      class="secondary-btn false"
    >
      Completed
    </button>
  </div>
</div>
`;
